packages/frontend/src/features/semanticViewer/runners/SemanticViewerResultsRunner.ts

```diff
@@ -96,91 +116,49 @@ export class SemanticViewerResultsRunner extends ResultsRunner {
         pivotLayoutOptions: VizPivotLayoutOptions[];
     } {
         return {
-            indexLayoutOptions: this.columns.reduce((acc, column) => {
-                const field =
-                    SemanticViewerResultsRunner.findSemanticLayerFieldFromColumn(
-                        this.fields,
-                        column.reference,
-                    );
-                if (field?.kind === FieldType.DIMENSION) {
-                    acc.push({
-                        reference: column.reference,
-                        type: this.getAxisType(column),
-                    });
-                }
-                return acc;
-            }, [] as VizIndexLayoutOptions[]),
-            valuesLayoutOptions: this.columns.reduce((acc, column) => {
-                const field =
-                    SemanticViewerResultsRunner.findSemanticLayerFieldFromColumn(
-                        this.fields,
-                        column.reference,
-                    );
-                if (field?.kind === FieldType.METRIC) {
-                    acc.push({
-                        reference: column.reference,
-                    });
-                }
-                return acc;
-            }, [] as VizValuesLayoutOptions[]),
-            pivotLayoutOptions: this.columns.filter((column) => {
-                const field =
-                    SemanticViewerResultsRunner.findSemanticLayerFieldFromColumn(
-                        this.fields,
-                        column.reference,
-                    );
-                return field?.kind === FieldType.DIMENSION;
-            }),
+            indexLayoutOptions: this.dimensions.map((dimension) => ({
+                reference: dimension.name,
+                type: getVizIndexTypeFromDimensionType(dimension.type),
+            })),
+            valuesLayoutOptions: this.metrics.map((metric) => ({
+                reference: metric.name,
+            })),
+            pivotLayoutOptions: this.dimensions.map((dimension) => ({
+                reference: dimension.name,
+                type: getVizIndexTypeFromDimensionType(dimension.type),
+            })),
```

Comment:
This allows us to select fields not present in the results (e.g. here I only selected `name` and `order total`)

![image](https://github.com/user-attachments/assets/e17ced86-602b-4c27-b709-50a268fc62b6)

We should filter them from the available columns

Reply:
Now being done in the constructor